### ISSUE TITLE
Real First Person 2's First Person Model Doesn't Scale with Your Size

### ISSUE THREAD

**Issue by Alexandqr** [necauqua/chiseled-me#83]
Hello, I'm back with another bug :D.
This is not necessarily a bug with this mod, or some kind of game breaking conflict with another mod. It's mainly a visual bug with Real First Person 2, where the body you see in first person mode doesn't actually scale with your player size.

To reproduce this behavior, just install both CM and RFP2 and change your size to anything that isn't 1, then look down.

This first person model should scale along with your size.

![2021-01-27_18 37 12](https://user-images.githubusercontent.com/72367464/106073130-16167780-60cf-11eb-9984-a40003f1a418.png)

Forge version I used was 1.12.2-14.23.5.2847.
Again, large modpack, but you only need the mod mentioned above along with CM.
The bug is really simple to reproduce, and it doesn't produce a log, so I haven't provided any logs to help.

This is obviously because RFP2's model isn't actually a part of the model seen in 3rd person, therefore isn't scaled, but still should have some kind of compatibility, so your body doesn't just detach from your head in first person.

I'm not a coder or anything, but I think this may be really easy to fix or really difficult, since all you need to do is add the scaling to the model RFP2 uses, but I'm not sure how difficult that is to do. Also, I think these two mods would look cool together, since you would be able to see yourself without going into F5.

**Comment by necauqua** [comment]
A bit of explanation for your assumptions:
I am changing the bytecode, so I can do whatever I want - and I wanted to put the scaling code around the model rendering, so _any_ model and literally anything the entity rendering code does gets scaled (for example I had to multiply guardian beam length by it's size because the beam is also getting scaled).

For it not to get scaled it must do some weird shit, which mods apprently tend to do (weird shit, that is) or maybe intercept the rendering code just like I do but for their purposes.

Well I'll take a look, although given it's not game-breaking it does not get the highest priority obviously.

**Comment by necauqua** [comment]
Oh wait I got to notice what mod you were talking about - well they are rendering their model in first person - not where entity models are usually rendered (in terms of code), so that explains why it is not scaled.

But yeah I'll do the compat thing, shouldn't be hard.

**Comment by necauqua** [comment]
Fixed in my local code that will be out (and maybe even released because current beta is very good) in like a week due to some circumstances

**Comment by Alexandqr** [comment]
Sounds great, can't wait to try it once it's released :D

**Comment by necauqua** [comment]
Well due to me trying to fix some omegaweird issues on discord the latest discord pinned beta also contains this fix, enjoy :)

**Comment by necauqua** [comment]
To clarify I expected to make an actual release with this fix and have no more betas, but oh well

**Comment by Alexandqr** [comment]
Ok I found out something new, and I'm pretty sure it's still related to this issue.
So I got my hands on the newest beta to try it out, and of course, I'm using a ton of different mods along with it. Whenever I try to start it up with RFP2, it just crashes, and it might be related to the fix you added creating more conflicts with some different mods. (Without RFP2, it runs fine, and it also runs fine without CM. Edited this in because I forgot to mention it)
Here's the log: [crash-2021-02-03_09.53.08-client.txt](https://github.com/necauqua/chiseled-me/files/5919825/crash-2021-02-03_09.53.08-client.txt)
